node/src/accountant/mod.rs

```diff
@@ -371,37 +366,31 @@ impl Handler<NodeFromUiMessage> for Accountant {
 }
 
 impl Accountant {
-    pub fn new(config: &mut BootstrapperConfig, dao_factories: DaoFactories) -> Accountant {
-        let payment_thresholds = Rc::new(
-            config
-                .payment_thresholds_opt
-                .take()
-                .expectv("Payment thresholds"),
-        );
-        let scan_intervals = config.scan_intervals_opt.take().expectv("Scan Intervals");
-        let when_pending_too_long_sec = config
-            .when_pending_too_long_opt
+    pub fn new(mut config: BootstrapperConfig, dao_factories: DaoFactories) -> Accountant {
+        let payment_thresholds = config
+            .payment_thresholds_opt
             .take()
-            .expectv("When pending too long sec");
-
+            .expectv("Payment thresholds");
+        let scan_intervals = config.scan_intervals_opt.take().expectv("Scan Intervals");
         let earning_wallet = Rc::new(config.earning_wallet.clone());
         let financial_statistics = Rc::new(RefCell::new(FinancialStatistics::default()));
+
         Accountant {
             scan_intervals,
-            suppress_initial_scans_opt: config.suppress_initial_scans_opt,
+            suppress_initial_scans: config.suppress_initial_scans,
             consuming_wallet: config.consuming_wallet_opt.clone(),
             earning_wallet: Rc::clone(&earning_wallet),
             payable_dao: dao_factories.payable_dao_factory.make(),
             receivable_dao: dao_factories.receivable_dao_factory.make(),
             pending_payable_dao: dao_factories.pending_payable_dao_factory.make(),
-            crashable: config.crash_point == CrashPoint::Message,
             scanners: Scanners::new(
```

Comment:
Actually, we can't pass `ado_factories` too early, as it is being used to make connections for DAOs inside the Accountant.

Reply:
I see. Still, nothing should prevent you from writing:

```
        let payable_dao =  dao_factories.payable_dao_factory.make();
        let receivable_dao = dao_factories.receivable_dao_factory.make(),
        let pending_payable_dao = dao_factories.pending_payable_dao_factory.make();

        let scanners = Scanners::new(
                dao_factories,
                payment_thresholds,
                Rc::new(payment_thresholds),
                Rc::clone(&earning_wallet),
                when_pending_too_long_sec,
                config.when_pending_too_long,
                Rc::clone(&financial_statistics),
            ); 

        Accountant {
            scan_intervals,
            suppress_initial_scans_opt: config.suppress_initial_scans_opt,
            suppress_initial_scans: config.suppress_initial_scans,
            consuming_wallet: config.consuming_wallet_opt.clone(),
            earning_wallet: Rc::clone(&earning_wallet),
            payable_dao,
            receivable_dao,
            pending_payable_dao,
            crashable: config.crash_point == CrashPoint::Message,
            scanners,
            crashable: config.crash_point == CrashPoint::Message,
            notify_later: NotifyLaterForScanners::default(),
            financial_statistics: Rc::clone(&financial_statistics),
            report_accounts_payable_sub_opt: None,
            retrieve_transactions_sub: None,
            request_transaction_receipts_subs_opt: None,
            report_new_payments_sub: None,
            report_sent_payments_sub: None,
            ui_message_sub: None,
            message_id_generator: Box::new(MessageIdGeneratorReal::default()),
            logger: Logger::new("Accountant"),
        }
```

However, you think about it. If you still like more your initial version I don't want to argue because of this. Of my liking, I think this more structuralized lay out will make reading easier.

Reply:
I find this easy to read. That's why I have made the change that you mentioned in the second iteration.

You may close this conversation as the intended change is done.